postcss/theme.js

```diff
@@ -166,13 +182,18 @@ module.exports = (options) => {
 
             root.prepend(lightRule, darkRule, darkMediaQuery);
 
-            for (const node of nodesToDelete)
+            for (const node of nodesToDelete) {
                 node.remove()
+            }
 
             // Remove all empty nodes
             root.each(node => {
-                if (node.nodes.length === 0) node.remove();
+                if (!node.nodes || node.nodes.length === 0) node.remove();
             })
+
+            // Reset global vars
+            rules = {};
```

Comment:
Does it maintain the state from file to file?! Because that would suck!

I bet this is what was actually causing our self referencing rules.

Just as one more annoying check, I wonder if this would fix all of our problems on it's own (without the global change). Would you mind trying? 

It just makes me feel a bit odd that it works in Storybook but not from another project.

Reply:
While not solving the same issue as solved by `useGlobal`, this does negate the need for `hasThemeAtRule`, so that's been removed.